src/taipy/core/_entity/_properties.py

```diff
@@ -35,10 +36,31 @@ def __setitem__(self, key, value):
                 tp.set(self._entity_owner)
                 _publish_event(*to_publish_event_parameters)
             else:
+                if key in self._pending_deletions:
+                    self._pending_deletions.remove(key)
                 self._pending_changes[key] = value
                 self._entity_owner._in_context_attributes_changed_collector.append(to_publish_event_parameters)
 
     def __getitem__(self, key):
         from taipy.config.common._template_handler import _TemplateHandler as _tpl
 
         return _tpl._replace_templates(super(_Properties, self).__getitem__(key))
+
+    def __delitem__(self, key):
+        super(_Properties, self).__delitem__(key)
+        from ... import core as tp
+
+        if hasattr(self, "_entity_owner"):
+            to_publish_event_parameters = [
+                _ENTITY_TO_EVENT_ENTITY_TYPE[self._entity_owner._MANAGER_NAME],
+                self._entity_owner.id,
+                EventOperation.DELETION,
+                key,
+            ]
```

Comment:
The DELETION EventOperation is designed to model an entity deletion, not an entity property deletion. I would use UPDATE with the attribute name equal to properties. We updated the properties of the entity by deleting an entry.

The attribute name in the __setitem__ method is also wrong. We updated the properties of the entity by changing an entry.

Does it make sense?

```suggestion
            to_publish_event_parameters = [
                _ENTITY_TO_EVENT_ENTITY_TYPE[self._entity_owner._MANAGER_NAME],
                self._entity_owner.id,
                EventOperation.UPDATE,
                'properties',
            ]
```

Reply:
yep I'm good with that :D I was hesitate to put it as deletion as well so 😃 oh the by that do you mean we should change the publish event parameter attribute name in setitem to "properties" instead of key?

Reply:
Yes